src/components/AssignmentCreatorForm.jsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import {t} from 'i18next';
+import classnames from 'classnames';
+import map from 'lodash-es/map';
+import chrono from 'chrono-node';
+
+import PropTypes from 'prop-types';
+import {Field, reduxForm} from 'redux-form/immutable';
+
+import {dateToString} from '../util/timeFormatter';
+
+import renderField from './renderField';
+import renderSelectField from './renderSelectField';
+
+function validate(values) {
+  const errors = {};
+  const input = values.get('date');
+  const parsedResponse = chrono.parse(input);
+
+  if (!values.get('course')) {
+    errors.course = '* Course Required';
+  } else if (values.get('course') === '') {
+    errors.course = 'Must Select a Course';
+  }
+  if (!values.get('date')) {
+    errors.date = '* Date Required';
+  }
+  if (parsedResponse[0]) {
+    const parsedDate = parsedResponse[0].start.date();
+    if (parsedDate < Date.now()) {
+      errors.date = 'Date must be in the future :)';
+    }
+  }
+  return errors;
+}
+
+function warn(values) {
+  const warnings = {};
+  const input = values.get('date');
+  const parsedResponse = chrono.parse(input);
+  if (parsedResponse[0]) {
+    const parsedDate = parsedResponse[0].start.date();
+    const dateString = dateToString(parsedDate);
+    warnings.date = dateString;
+  }
+  return warnings;
```

Comment:
OK so this is a clever use of `warnings`, but of course this is not _actually_ a warning so it does feel a little funny. I have a have/eat cake proposal with the caveat that I have much less experience (none) working with `redux-form` than you, so my apologies if this is unrealistic and/or you’ve already tried it.

Basic idea: The form should have both a `natural_language_date` field and a `date` field. The `natural_language_date` field holds the string that the user typed, and the `date` field holds the translated version.

The `natural_language_date` field would be rendered as it is now, a normal text input. The `date` field would be rendered as a `readonly` input, styled to not look like an input at all.

Using an `onChange` handler on the `natural_language_date` `<Field>` component, we’d register a callback that calls [`change()`](https://redux-form.com/7.4.2/docs/api/props.md/#-code-change-field-string-value-any-function-code-) to update the `date` field accordingly (i.e. using `chrono.parse`).

The advantage here is that, if I’m not mistaken, nothing else would need to worry about the `natural_language_date` input, `chrono.parse` etc.—everything else (including the validation etc.) could just look at the already-parsed value of `date`.

(Might be even slicker to let `date` itself flow through the `parse`/`format` flow, not sure if that would make the code simpler or more complicated though)